### DISCUSSION TITLE
Provide a way for Admin to block problematic URLs

### DISCUSSION THREAD

**Discussion by michael-tharrington** [forem/forem#18477] [Feature Requests]
[Andy Piper](https://dev.to/andypiper), a super helpful community member and moderator at DEV, suggested that we come up with a way to add bad URLs to a denylist so that we can block spammers from continuously signing up and sharing the same URLs in posts, their profiles, & comments.

Spammers are most frequently joining our communities to try to publicize and boost SEO for specific websites via backlinking. If we could prevent them from gaining any traction here, we might discourage them from posting on our communities altogether.

I can think of at least 2 solid ways we could approach this. For both ideas we'd need to have a place in the admin section (my suggestion would be in `/admin/customization/config` > Rate limits and anti-spam) where admin can add problematic URLs to a denylist, then:

1) We could prevent folks from including these problematic URLs in their posts/comments/listings/profiles altogether. Perhaps when they try to post a problematic URL, they'd be hit with an error message. BONUS: they'd be auto-flagged to admin in the Abuse Reports section (i.e. `/admin/moderation/feedback_messages`)

2) Whenever someone uses one of these URLs, we could suspend their account and unpublish all of their content — posts, comments, etc. Again, I think it would be ideal to auto-flag these accounts to admin as well. This way we can decide if we need to take any further action, like banishing them for instance.

**Comment by benhalpern** [comment]
I think this makes a lot of sense.

Unlike problematic _phrases_ where there is _way more nuance_ as to whether a phrase is spam, and you wouldn't want a false negative.

There are some solid ideas here. Some questions about whether we want to make this solid permanent app behavior or _maybe_ this is third party logic the API can help enable. But there are definitely good reasons to keep this _in app_. We'll take this into immediate consideration.

Thanks for bringing this up!
